src/NetAnalyzers/UnitTests/Microsoft.NetCore.Analyzers/Performance/AvoidSingleUseOfLocalJsonSerializerOptionsTests.cs

```diff
@@ -0,0 +1,574 @@
+
+        [Fact]
+        public Task CS_UseNewLocalOptionsAsArgument_SecondLocalReference()
+            => VerifyCS.VerifyAnalyzerAsync(@"
+using System.Text.Json;
+
+class Program
+{
+    static string Serialize<T>(T value)
+    {
+        JsonSerializerOptions opt = new JsonSerializerOptions();
+        _ = opt;
+
+        return JsonSerializer.Serialize(value, opt);
+    }
+}
+", VerifyCS.Diagnostic(AvoidSingleUseOfLocalJsonSerializerOptions.s_Rule).WithSpan(8, 37, 8, 64));
+
+        [Fact] // this could be better handled with data flow analysis.
+        public Task CS_UseNewLocalOptionsAsArgument_OverwriteLocal()
+    => VerifyCS.VerifyAnalyzerAsync(@"
+using System.Text.Json;
+
+class Program
+{
+    static JsonSerializerOptions s_options;
+
+    static string Serialize<T>(T value)
+    {
+        JsonSerializerOptions opt = new JsonSerializerOptions();
+        opt = s_options;
+
+        return JsonSerializer.Serialize(value, opt);
+    }
+}
+", VerifyCS.Diagnostic(AvoidSingleUseOfLocalJsonSerializerOptions.s_Rule).WithSpan(10, 37, 10, 64));
+
+        [Theory]
+        [InlineData("opt1")]
+        [InlineData("opt2")]
+        public Task CS_UseNewLocalOptionsAsArgument_MultiAssignment(string expression)
+            => VerifyCS.VerifyAnalyzerAsync($@"
+using System.Text.Json;
+
+class Program
+{{
+    static string Serialize<T>(T value)
+    {{
+        JsonSerializerOptions opt1, opt2;
+        opt1 = opt2 = new JsonSerializerOptions();
+
+        return JsonSerializer.Serialize(value, {expression});
+    }}
+}}
+", VerifyCS.Diagnostic(AvoidSingleUseOfLocalJsonSerializerOptions.s_Rule).WithSpan(9, 23, 9, 50));
+
+        [Fact]
+        public Task CS_UseNewLocalOptionsAsArgument_Delegate()
+    => VerifyCS.VerifyAnalyzerAsync(@"
+using System;
+using System.Text.Json;
+
+class Program
+{
+    static Action Serialize<T>(T value)
+    {
+        Action lambda = () =>
+        {
+            JsonSerializerOptions opt = new JsonSerializerOptions();
+            JsonSerializer.Serialize(value, opt);
+        };
+        return lambda;
+    }
+}
+", VerifyCS.Diagnostic(AvoidSingleUseOfLocalJsonSerializerOptions.s_Rule).WithSpan(11, 41, 11, 68));
+
+        [Fact]
+        public Task CS_UseNewLocalOptionsAsArgument_LocalFunction()
+=> VerifyCS.VerifyAnalyzerAsync(@"
+using System.Text.Json;
+
+class Program
+{
+    static string Serialize<T>(T value)
+    {
+        return LocalFunc();
+
+        string LocalFunc()
+        {
+            JsonSerializerOptions opt = new JsonSerializerOptions();
+            return JsonSerializer.Serialize(value, opt);
+        }
+    }
+}
+", VerifyCS.Diagnostic(AvoidSingleUseOfLocalJsonSerializerOptions.s_Rule).WithSpan(12, 41, 12, 68));
+
+        #endregion
+
+        #region No Diagnostic Tests
+        [Fact]
+        public Task CS_UseNewOptionsAsArgument_NonSerializerMethod_NoWarn()
+            => VerifyCS.VerifyAnalyzerAsync(@"
```

Comment:
For my education, the `VerifyAnalyzerAsync` overload that only takes a source parameter will fail the test if _any_ diagnostics have been emitted?

Reply:
Yup.